src/heartbeat.ts

```diff
@@ -76,8 +76,13 @@ export class HeartbeatManager extends Disposable {
         this._register(vscode.window.onDidCloseTerminal(() => this.updateLastActivity('onDidCloseTerminal')));
         this._register(vscode.window.onDidChangeTerminalState(() => this.updateLastActivity('onDidChangeTerminalState')));
         this._register(vscode.window.onDidChangeWindowState((e) => {
-            this.focused = e.focused;
-            this.updateLastActivity('onDidChangeWindowState');
+            // active property was introduced in 1.89, but not exist in 1.75
+            // https://code.visualstudio.com/api/references/vscode-api#WindowState
+            const isActive: Boolean | undefined = (e as any).active;
+            if (isActive === true || e.focused !== this.focused) {
+                this.focused = e.focused;
+                this.updateLastActivity('onDidChangeWindowState');
+            }
```

Comment:
With this new api we don't need the other events so let remove them, and send heartbeat only when `isActive` is `true` every `HEARTBEAT_INTERVAL + random value`